quipucords/api/scan/tests_scan.py

```diff
@@ -644,30 +644,27 @@ def setUp(self):
             become_password=None,
             ssh_keyfile=None,
         )
-        self.cred.save()
         self.cred_for_upload = self.cred.id
 
-        self.source = Source(name="source1", source_type="network", port=22)
-        self.source.save()
+        self.source = Source.objects.create(
+            name="source1", source_type="network", port=22
+        )
         self.source.credentials.add(self.cred)
-        self.source.save()
 
-        self.test1 = Scan(name="test1", scan_type=ScanTask.SCAN_TYPE_INSPECT)
-        self.test1.save()
+        self.test1 = Scan.objects.create(
+            name="test1", scan_type=ScanTask.SCAN_TYPE_INSPECT
+        )
         self.test1.sources.add(self.source)
-        self.test1.save()
```

Comment:
don't we need the saves after adding the Many to Many relationships ? saw examples here https://docs.djangoproject.com/en/4.2/topics/db/examples/many_to_many/ doing the save.

Reply:
https://docs.djangoproject.com/en/4.2/ref/models/relations/

> Note that add(), aadd(), create(), acreate(), remove(), aremove(), clear(), aclear(), set(), and aset() all apply database changes immediately for all types of related fields. In other words, there is no need to call save()/asave() on either end of the relationship.

🪄 ✨

Reply:
FWIW, that `Note` was relatively new information for me too. I only somewhat recently found that comment, and I've been taking advantage of that knowledge for cleanups like this. I wonder if that automatic save was not always true for older versions of Django (I've been Django-ing since at least 1.x…), but it does seem to work consistently now.

Reply:
LOL… well, that note has been there since the oldest version of the docs for Django 1.8. I guess I just never noticed it until recently! https://docs.djangoproject.com/en/1.8/ref/models/relations/

Reply:
Ahh, thank you for that Note.  Could have been a new addition to Django, and the examples I saw were not updated accordingly.

Reply:
At least for create() this hasnt changed. 
https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.create
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#django.db.models.query.QuerySet.create